File: ois/ois20/round4/ois11_round4_solutions.tar(1)/gameshow.pas
{ just simulates the process }

{ constraints }
const
    MAXN = 100000;

{ input data }
var
    N, M, K, i : longint;
    P, C       : array[0..MAXN-1] of longint;

begin
{
    uncomment the following lines if you want to read/write from files
    assign(input,  'input.txt');  reset(input);
    assign(output, 'output.txt'); rewrite(output);
}

    readln(N, M);
    for i:=0 to N-1 do
        read(P[i]);
    readln();
    for i:=0 to N-1 do
        read(C[i]);
    readln();

    K := 0;
    for i:=0 to N-1 do
        if P[i] > M then M := M + C[i]
        else begin
            M := M - P[i];
            K := K + 1;
        end;

    writeln(K, ' ', M); { print result }
end.

File: ois/ois20/round4/ois11_round4_solutions.tar(1)/gameshow.cpp
/*
 * This template is valid both in C and in C++,
 * so you can expand it with code from both languages.
 * NOTE: it is recommended to use this even if you don't
 * understand the following code.
 */

#include <stdio.h>
#include <assert.h>

// constraints
#define MAXN 100000

// input data
int N, M, i;
int P[MAXN], C[MAXN];

int main() {
//  uncomment the following lines if you want to read/write from files
//  freopen("input.txt", "r", stdin);
//  freopen("output.txt", "w", stdout);

    assert(2 == scanf("%d %d", &N, &M));
    for(i=0; i<N; i++)
        assert(1 == scanf("%d", &P[i]));
    for(i=0; i<N; i++)
        assert(1 == scanf("%d", &C[i]));

    // insert your code here
    int count = 0;
    for (int i = 0; i < N; i++) {
        if (M >= P[i]) {
            count++;
            M -= P[i];
        } else {
            M += C[i];
        }
    }
    
    printf("%d %d\n", count, M); // print the result
    return 0;
}
